pynxtools/dataconverter/convert.py

```diff
@@ -72,8 +72,12 @@ def convert(input_file: Tuple[str],
             generate_template: bool = False,
             fair: bool = False,
             undocumented: bool = False,
+            logger_: logging.Logger = None,
```

Comment:
The best practice for logging is different. The docs recommend to use a modul level logger with a name, e.g., creating `logger` variable at the start of the file (see https://docs.python.org/3/howto/logging.html#advanced-logging-tutorial):

```python
import logging

logger = logging.getLogger(__file__)
```

Then you can just use `logger.warning("My warning")` inside the file. Also use `%s` syntax when formatting the string, e.g., `logger.warning("Invalid value: %s", my_var)` when logging my_var, because logging does not evaluate my_var in case the logging level is disabled.

This allows another tool using pynxtools to hook into the logger with our name and display the emitted logs. I don't know if we have to follow naming conventions for nomad (I had this for uvicorn where I needed to create a name like `logging.getLogger(f'uvicorn.{__file__}')` and then everything was displayed automagically).

Reply:
I will try to follow it.

Reply:
Yah, when you mentioned about `logger` warning in this, I also thought to have something like this as you mentioned here. I see the beauty of logger.

Reply:
One thing, `logger.warning("Invalid value: %s", my_var)` does not replace %s when in nomad logger. But the same syntax works in console when calling reader.

Reply:
Module level logger in pynxtools is also working nicely.

Reply:
> One thing, `logger.warning("Invalid value: %s", my_var)` does not replace %s when in nomad logger. But the same syntax works in console when calling reader.

Ok, this is odd. Normally, this is recommended (see https://docs.python.org/3/howto/logging.html#optimization). Maybe this behaviour is deactivated for some reason in nomad, because there I just see f-strings being used (and the pylint errors are also deactivated). So we can just follow the nomad way: use f-strings and add `logging-not-lazy`, `logging-format-interpolation`, `logging-fstring-interpolation` to `.pylintrc`. But I would be interested why it is like this in nomad